File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/Delphi/Invoice/InvoiceDemoMain.pas
unit InvoiceDemoMain;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, cxGraphics, cxControls, cxLookAndFeels, cxLookAndFeelPainters,
  dxSkinsCore, dxSpreadSheetCore, dxSpreadSheetFunctions,
  dxSpreadSheetGraphics, dxSpreadSheetClasses, dxSpreadSheetTypes,
  dxSpreadSheet, BaseForm, dxCore, dxCoreClasses, dxHashUtils, dxSpreadSheetCoreHistory, dxSpreadSheetCoreStyles,
  dxSpreadSheetCoreStrs, dxSpreadSheetConditionalFormatting, dxSpreadSheetConditionalFormattingRules,
  dxSpreadSheetContainers, dxSpreadSheetFormulas, dxSpreadSheetHyperlinks, dxSpreadSheetStyles, dxSpreadSheetPrinting,
  dxSpreadSheetUtils, cxSplitter, dxSpreadSheetFormulaBar;

type
  TfrmInvoice = class(TfmBaseForm)
    SpreadSheet: TdxSpreadSheet;
    FormulaBar: TdxSpreadSheetFormulaBar;
    FormulaBarSplitter: TcxSplitter;
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    function GetSpreadSheet: TdxSpreadSheet; override;
  end;

var
  frmInvoice: TfrmInvoice;

implementation

{$R *.dfm}

procedure TfrmInvoice.FormCreate(Sender: TObject);
begin
  inherited;
  SpreadSheet.LoadFromFile('..\..\Data\Invoice.xlsx');
  WindowState := wsMaximized;
end;

function TfrmInvoice.GetSpreadSheet: TdxSpreadSheet;
begin
  Result := SpreadSheet;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/CBuilder/Invoice/InvoiceDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "InvoiceDemoMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BaseForm"
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "dxCore"
#pragma link "dxCoreClasses"
#pragma link "dxHashUtils"
#pragma link "dxSpreadSheet"
#pragma link "dxSpreadSheetClasses"
#pragma link "dxSpreadSheetCore"
#pragma link "dxSpreadSheetFormulas"
#pragma link "dxSpreadSheetFunctions"
#pragma link "dxSpreadSheetGraphics"
#pragma link "dxSpreadSheetTypes"
#pragma link "dxSpreadSheetFormulaBar"
#pragma link "cxSplitter"
#pragma resource "*.dfm"
TfrmInvoice *frmInvoice;
//---------------------------------------------------------------------------
__fastcall TfrmInvoice::TfrmInvoice(TComponent* Owner)
	: TfmBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmInvoice::FormCreate(TObject *Sender)
{
  TfmBaseForm::FormCreate(Sender);
  SpreadSheet->LoadFromFile("..\\..\\Data\\Invoice.xlsx");
  WindowState = wsMaximized;
}
//---------------------------------------------------------------------------

TdxSpreadSheet* TfrmInvoice::GetSpreadSheet()
{
  return SpreadSheet;
}
//---------------------------------------------------------------------------

